static/src/javascripts/projects/common/modules/commercial/user-ad-targeting.js

```diff
@@ -1,11 +1,12 @@
 // @flow
 import { local } from 'lib/storage';
 import { getUserFromCookie, getUserFromApi } from 'common/modules/identity/api';
+import { consentState } from 'lib/cmp';
 
 const userSegmentsKey = 'gu.ads.userSegmentsData';
 
-const getUserSegments = function(): Array<any> {
-    if (local.isAvailable()) {
+const getUserSegments = (): Array<any> => {
+    if (local.isAvailable() && consentState('advertisement') !== false) {
         let userCookieData;
```

Comment:
can we inline this with line 14 as below:
let userCookieData = getUserFromCookie();

Reply:
Good catch!